thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js

```diff
@@ -305,140 +306,153 @@ export default Controller.extend({
-
-      //
-      // breakdowns
-      //
-      if (activeTab === ROOTCAUSE_TAB_DIMENSIONS) {
-        const metricUrns = new Set(filterPrefix(context.urns, 'thirdeye:metric:'));
-        const currentUrns = [...metricUrns].map(toCurrentUrn);
-        const baselineUrns = [...metricUrns].map(toBaselineUrn);
-        const sizeMetricCurrentUrns = [...sizeMetricUrns].map(toCurrentUrn);
-        breakdownsService.request(context, new Set(currentUrns.concat(baselineUrns).concat(sizeMetricCurrentUrns)));
-      }
+    //
+    // timeseries
+    //
+    const timeseriesUrns = new Set(selectedUrns);
 
-      //
-      // scores
-      //
-      if (activeTab === ROOTCAUSE_TAB_METRICS) {
-        const scoresUrns = new Set(relatedMetricUrns);
-        scoresService.request(context, new Set(scoresUrns));
-      }
+    if (activeTab === ROOTCAUSE_TAB_TREND) {
+      [...relatedMetricUrns].forEach((urn) => {
+        timeseriesUrns.add(toCurrentUrn(urn));
+        timeseriesUrns.add(toBaselineUrn(urn));
+      });
+    }
 
-      //
-      // aggregates
-      //
-      const offsets = ['current', 'baseline', 'wo1w', 'wo2w'];
-      const offsetUrns = [...relatedMetricUrns]
-        .map(urn => [].concat(offsets.map(offset => toOffsetUrn(urn, offset))))
-        .reduce((agg, l) => agg.concat(l), []);
-
-      const anomalyOffsets = ['current', 'baseline', 'wo1w', 'wo2w', 'wo3w', 'wo4w'];
-      const anomalyOffsetUrns = [...anomalyMetricUrns]
-        .map(urn => [].concat(anomalyOffsets.map(offset => toOffsetUrn(urn, offset))))
-        .reduce((agg, l) => agg.concat(l), []);
-
-      //
-      // call graph
-      //
-      if (activeTab === ROOTCAUSE_SERVICE_CALLGRAPH) {
-        callgraphService.request(context, [...context.urns]);
-      }
+    timeseriesService.request(context, timeseriesUrns);
+
+    //
+    // anomaly function baselines
+    //
+    const anomalyFunctionUrns = filterPrefix(context.anomalyUrns, 'thirdeye:event:anomaly');
+    anomalyFunctionService.request(context, new Set(anomalyFunctionUrns));
+
+    //
+    // breakdowns
+    //
+    if (activeTab === ROOTCAUSE_TAB_DIMENSIONS) {
+      const metricUrns = new Set(filterPrefix(context.urns, 'thirdeye:metric:'));
+      const currentUrns = [...metricUrns].map(toCurrentUrn);
+      const baselineUrns = [...metricUrns].map(toBaselineUrn);
+      const sizeMetricCurrentUrns = [...sizeMetricUrns].map(toCurrentUrn);
+      breakdownsService.request(context, new Set(currentUrns.concat(baselineUrns).concat(sizeMetricCurrentUrns)));
+    }
 
-      aggregatesService.request(context, new Set([...offsetUrns, ...anomalyOffsetUrns]));
+    //
+    // scores
+    //
+    if (activeTab === ROOTCAUSE_TAB_METRICS) {
+      const scoresUrns = new Set(relatedMetricUrns);
+      scoresService.request(context, new Set(scoresUrns));
+    }
 
+    //
+    // aggregates
+    //
+    const offsets =
+      context.compareMode === 'forecast'
+        ? ['current', 'baseline', 'yoy', 'upper', 'lower']
```

Comment:
Will the `upper` and `lower` make separate calls to get the boundary values? I would prefer to make the boundary values be returned with the forecast values in one call because they are generated by the forecast provider together. I think it is quite expensive to run it 3 times.

Reply:
Yes, the endpoint should grab all three values at once.  I wanted to send them to the frontend in this format to stay consistent with how the metric table currently receives its values.

Reply:
The implementation is in this PR: https://github.com/apache/incubator-pinot/pull/6432